src/gz/watchlist.c

```diff
@@ -262,13 +269,41 @@ static int destroy_proc(struct menu_item *item)
   return 0;
 }
 
+static int toggle_visibility_proc(struct menu_item *item,
+                      enum menu_callback_reason reason,
+                      void *data)
+{
+  if (reason == MENU_CALLBACK_SWITCH_ON)
```

Comment:
I suggest enclosing all of this code, except the last two lines, in a `MENU_CALLBACK_CHANGED` instead, and using `menu_checkbox_get(item)` as the new value for the setting. Enabling or disabling the watches on every callback is not necessary.

Reply:
The reason I didn't do this was because I need this callback to fire when `command_togglewatches` runs. If I only ever enable/disable menus on change, I'll miss the times where the settings value is updated from the command. I feel like maybe I'm abusing the "think" mechanic here? Open to suggestions on how to do this better.

Reply:
Ah, I see. In that case I suggest adding a new method for the watchlist, something like `watchlist_show/hide_watches` or `watchlist_set_visible`, which would be called from both `command_togglewatches` and `toggle_visibility_proc`.

Reply:
Done! Looks so much better now. Thank you! 😄